### PR TITLE
Add support for primary keys that have a relation

### PR DESCRIPTION
**Pull request by vdboor**
This addresses the needs for the dataset 'aardgasverbruik', that has a primary key that is also a relationship.

### CODE REVIEW

File: src/dso_api/dynamic_api/serializers/base.py [hunk 1]
```diff
@@ -545,6 +545,20 @@ class LinkSerializer(FieldAccessMixin, DSOModelSerializerBase):
     This applies the field permissions to the display_name field (through FieldAccessMixin).
     """
 
+    def build_standard_field(self, field_name, model_field):
+        field_class, field_kwargs = super().build_standard_field(field_name, model_field)
+        if model_field.one_to_one and model_field.primary_key:
+            # When the OneToOneField is a primary key, and field isn't statically declared,
+            # DRF will generate one where on the fly but complain about a missing "view_name".
```
Comment (jjmurre) [inline src/dso_api/dynamic_api/serializers/base.py:552]: typo: where -> when ?

File: src/tests/files/datasets/aardgasverbruik.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  "id": "aardgasverbruik",
+  "title": "aardgasverbruik",
+  "status": "beschikbaar",
+  "version": "1.0.0",
+  "owner": "Liander / Gemeente Amsterdam",
+  "publisher": "Datateam Basisstatistiek",
+  "crs": "EPSG:28992",
+  "creator": "Onderzoek en Statistiek",
+  "auth": "OPENBAAR",
+  "authorizationGrantor": "basisstatistiek.os@amsterdam.nl",
```
[the remark this answers is not in the document]
Comment (vdboor) [reply src/tests/files/datasets/aardgasverbruik.json]: good point!

File: src/dso_api/dynamic_api/serializers/base.py (hunk 1 above)
Comment (vdboor) [reply src/dso_api/dynamic_api/serializers/base.py]: tx, done!

**Comment by lbam** [comment]
Jenkins is not happy:

```
=================================== FAILURES ===================================
___________ TestDynamicSerializer.test_loose_relation_serialization ____________

drf_request = <rest_framework.request.Request: GET '/v1/dummy/'>
statistieken_model = <SimpleLazyObject: <class 'dso_api.dynamic_api.meldingen.models.statistieken'>>

    @staticmethod
    def test_loose_relation_serialization(drf_request, statistieken_model):
        """Prove that the serializer factory generates the right link
        for a loose relation field
        """
        statistiek = statistieken_model.objects.create(
            id=1,
            buurt_id="03630000000078",
        )
        StatistiekenSerializer = serializer_factory(statistieken_model)
    
        statistieken_serializer = StatistiekenSerializer(
            statistiek, context={"request": drf_request}
        )
        buurt_field = statistieken_serializer.fields["_links"].fields["buurt"]
>       assert buurt_field.__class__.__name__ == "GebiedenBuurtenLooseLinkSerializer"
E       AssertionError: assert 'GebiedenBuur...ierSerializer' == 'GebiedenBuur...inkSerializer'
E         - GebiedenBuurtenLooseLinkSerializer
E         ?                ^^^^ ^ ^^
E         + GebiedenBuurtenRawIdentifierSerializer
E         ?                ^^^^^ ^^ ^^^^
```

**Review by jjmurre — APPROVED** [review]
LGTM. Didn't know `django_assert_num_queries`, nice.

File: src/tests/test_dynamic_api/test_serializers.py [hunk 3]
```diff
@@ -313,6 +313,52 @@ def test_dataset_path(
...
+        PostcodeRange.objects.create(
+            id=MraLiander.objects.create(id="123"), gemiddeld_verbruik=200
+        )
+
+        # Prove that the serializer can be constructed
+        PostcodeRangeSerializer = serializer_factory(PostcodeRange)
+
+        # Also prove that the optimizations work and the underlying model
+        # doesn't have to be fetched to render the _links.id field:
+        with django_assert_num_queries(1):
```
Comment (lbam) [inline src/tests/test_dynamic_api/test_serializers.py:332]: Nice!

**Review by lbam — APPROVED** [review]

**Pull request merged**